### PR TITLE
Reorganizing GUI

### PR DESCRIPTION
**Pull request by jsignell**
Reorganizing GUI around the concept of a separation between catalog actions and sources actions after discussion with @jbednar and @jlstevens

Screen Shot 2019-04-18 at 4 57 04 PM

### CODE REVIEW

**Comment by jbednar** [comment]
Is the extra space after "Catalogs" intentional?  To me it seems like there should be less...

**Comment by martindurant** [comment]
This seems like a reasonable layout change.
Note that, as the GUI evolves and gains functionality, we will likely have *several* layout changes...
Thought: is it possible to introduce colour/bold into the source description to highlight the keys like "container:"?

**Comment by jsignell** [comment]
> Is the extra space after "Catalogs" intentional?

That space snuck into the bokeh css recently. Not sure how to get rid of it, but I'll discuss with Philipp

>  is it possible to introduce colour/bold into the source description to highlight the keys

I am rendering it was straight str right now, but I guess I could use markdown if there were syntax highlighting in the markdown library that panel depends on.

**Comment by jbednar** [comment]
https://python-markdown.github.io/extensions/code_hilite/

**Comment by philippjfr** [comment]
I'd like to see a panel issue asking to expose the markdown extensions so code highlighting could easily be enabled by a user.

**Comment by jsignell** [comment]
TO DO:
 
- [ ] Serialize/dehydrate and rehydrate methods
- [ ] Docs
- [ ] Write tests to ensure that panel isn't necessary (GUI isn't fully instantiated until display)

**Comment by jsignell** [comment]
If this passes CI, I'd like to merge. Sound good @martindurant?
 
Screen Shot 2019-04-24 at 1 46 59 PM

**Comment by martindurant** [comment]
Any idea why the left three buttons are not *quite* level with the plot button in the image?

File: docs/source/gui.rst [hunk 1]
```diff
@@ -1,100 +1,149 @@
...
+source entries from the catalog in the **list of sources**.
 
-  -  Search: opens a sub-panel for finding entries in the currently-selected catalog (and it's
-     sub-catalogs)
+Below the **lists of catalogs** is a row of buttons that are used for adding, removing and
+searching-within catalogs:
 
-  -  Add: opens a sub-panden for adding catalogs to the interface, by either browsing for a local
-     YAML file or by entering a URL for a catalog, which can be a remote file or Intake server
+-  **Search**: opens a sub-panel for finding entries in the currently-selected catalog (and it's
```
Comment (martindurant) [inline docs/source/gui.rst]: Should be in the order in the UI (which I think is the right order): add, remove, search. The sections below will need to be shuffled too.

**Comment by jsignell** [comment]
> Any idea why the left three buttons are not quite level with the plot button in the image?

Honestly no and it's been bothering me. It isn't always like that. There is a very mild correction that happens when you interact, and then it sticks.

File: docs/source/gui.rst (hunk 1 above)
Comment (jsignell) [reply docs/source/gui.rst]: Do you think the section titles should just be "Add", "Remove", and "Search"?

**Comment by martindurant** [comment]
For the sake of not instantiating things until required, what do you make of the following
```diff
--- a/intake/catalog/base.py
+++ b/intake/catalog/base.py
@@ -25,7 +25,6 @@ from .remote import RemoteCatalogEntry
 from .utils import flatten, reload_on_change, RemoteCatalogError
 from ..source.base import DataSource
 from ..compat import unpack_kwargs
-from .gui import CatalogGUI
 logger = logging.getLogger('intake')


@@ -114,7 +113,7 @@ class Catalog(DataSource):
         self.updated = time.time()
         self._entries = self._make_entries_container()
         self.force_reload()
-        self._gui = CatalogGUI(cat=self, visible=False)
+        self._gui = None

     @classmethod
     def from_dict(cls, entries, **kwargs):
@@ -415,6 +414,9 @@ class Catalog(DataSource):

     @property
     def gui(self):
+        from .gui import CatalogGUI
+        if self._gui is None:
+            self._gui = CatalogGUI(cat=self, visible=False)
         self._gui.visible = True
         return self._gui
```

**Comment by jsignell** [comment]
@martindurant isn't that basically what I have now?

**Comment by martindurant** [comment]
No, the widget is instantiated for every cat instance immediately. I am suggesting that instantiation shouldn't happen until the widget is requested. The user API doesn't change.

**Comment by jsignell** [comment]
That is what is on master. But this PR already made the change to instantiate only on request.

**Comment by martindurant** [comment]
Sorry, I seem to have failed to fetch... forget it.

**Comment by jsignell** [comment]
Ok for me to merge @martindurant?

**Pull request merged**